azure-iot-device/azure/iot/device/iothub/sync_handler_manager.py

```diff
@@ -0,0 +1,232 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module contains the manager for handler methods used by the callback client"""
+
+import logging
+import threading
+import abc
+import six
+from azure.iot.device.common import handle_exceptions
+from azure.iot.device.common.chainable_exception import ChainableException
+from azure.iot.device.iothub.sync_inbox import InboxEmpty
+import concurrent.futures
+
+logger = logging.getLogger(__name__)
+
+MESSAGE = "_on_message_received"
+METHOD = "_on_method_request_received"
+TWIN_DP_PATCH = "_on_twin_desired_properties_patch_received"
+# TODO: add more for "event"
+
+
+class HandlerManagerException(ChainableException):
+    pass
+
+
+class HandlerRunnerKillerSentinel(object):
+    pass
+
+
+@six.add_metaclass(abc.ABCMeta)
+class AbstractHandlerManager(object):
+    """Partial class that defines handler manager functionality shared between sync/async"""
+
+    def __init__(self, inbox_manager):
+        self._inbox_manager = inbox_manager
+
+        self._handler_runners = {
```

Comment:
>_handler_runners [](start = 13, length = 16)

I think this needs a better name.  "handler runner" sounds like to words that mean almost the same thing pasted together.  I don't know what that name would be.  handler_invocation_object?  words are not working for me right now.

Reply:
I'm open to an alternative, but I've yet to think of something better. The thing is it's not an object - it's a function/coroutine that loops, calling the handler. i.e. it is a runner for a handler.

---
In reply to: [473419040](https://github.com/Azure/azure-iot-sdk-python/pull/644#discussion_r473419040) [](ancestors = 473419040)

Reply:
Let's drop it for now.  I can't think of anything better.

---
In reply to: [473481960](https://github.com/Azure/azure-iot-sdk-python/pull/644#discussion_r473481960) [](ancestors = 473481960,473419040)